seed/utils/api_schema.py

```diff
@@ -97,20 +98,37 @@ def body_field(cls, required, description, name='body', params_to_formats={}):
         )
 
     @classmethod
-    def schema_factory(cls, params_to_formats):
-        """Translates simple dictionary into an openapi Schema instance
+    def schema_factory(cls, obj, **kwargs):
+        """Translates an object into an openapi Schema
 
-        :param params_to_formats: dict[str, str]
+        :param obj: str, list, dict[str, obj]
         :return: drf_yasg.openapi.Schema
         """
-        return openapi.Schema(
-            type=openapi.TYPE_OBJECT,
-            properties={
-                k: cls.body_parameter_formats[format_name]
-                for k, format_name
-                in params_to_formats.items()
-            }
-        )
+        if type(obj) is str:
+            if obj not in cls.body_parameter_formats:
+                raise Exception(f'Invalid type "{obj}"; expected one of {cls.body_parameter_formats.keys()}')
+            return cls.body_parameter_formats[obj]
+
+        if type(obj) is list:
+            if len(obj) != 1:
+                raise Exception('List types must have exactly one element to specify the schema of `items`')
+            return openapi.Schema(
+                type=openapi.TYPE_ARRAY,
+                items=cls.schema_factory(obj[0])
+            )
```

Comment:
That's right, this can now create the arrays. Good catch, I can go back and use `['integer']` instead of `'integer_array'` and delete those array types from the class